io_scene_nif/modules/property/texture/loader/texture_loader.py

```diff
@@ -66,143 +67,142 @@ def get_texture_hash(self, source):
+            b_image = bpy.ops.image.open(tex)
+            # Blender will return an image object even if the file format is not supported,
+            # so to check if the image is actually loaded an error is forced via "b_image.size"
+            try:
+                b_image.size
+            except:  # RuntimeError: couldn't load image data in Blender
+                b_image = None  # not supported, delete image object
+        finally:
+            stream.close()
+
         return [fn, b_image]
 
-    def import_source(self, source):
+    def import_external_source(self, source):
         b_image = None
         fn = None
 
-        # the n_texture uses an external image file
+        # the texture uses an external image file
         if isinstance(source, NifFormat.NiSourceTexture):
             fn = source.file_name.decode()
         elif isinstance(source, str):
             fn = source
         else:
             raise TypeError("source must be NiSourceTexture or str")
+
         fn = fn.replace('\\', os.sep)
         fn = fn.replace('/', os.sep)
-
         # go searching for it
-        importpath = os.path.dirname(NifOp.props.filepath)
-        searchPathList = [importpath]
+        import_path = os.path.dirname(NifOp.props.filepath)
+        search_path_list = [import_path]
         if bpy.context.user_preferences.filepaths.texture_directory:
-            searchPathList.append(bpy.context.user_preferences.filepaths.texture_directory)
+            search_path_list.append(
+                bpy.context.user_preferences.filepaths.texture_directory)
 
-        # TODO [n_texture] Implement full n_texture path finding.
+        # TODO [general][path] Implement full texture path finding.
         nif_dir = os.path.join(os.getcwd(), 'nif')
-        searchPathList.append(nif_dir)
+        search_path_list.append(nif_dir)
 
-        # if it looks like a Morrowind style path, use common sense to guess n_texture path
-        meshes_index = importpath.lower().find("meshes")
+        # if it looks like a Morrowind style path, use common sense to guess texture path
+        meshes_index = import_path.lower().find("meshes")
         if meshes_index != -1:
-            searchPathList.append(importpath[:meshes_index] + 'textures')
+            search_path_list.append(import_path[:meshes_index] + 'textures')
 
-        # if it looks like a Civilization IV style path, use common sense to guess n_texture path
-        art_index = importpath.lower().find("art")
+        # if it looks like a Civilization IV style path, use common sense to guess texture path
+        art_index = import_path.lower().find("art")
         if art_index != -1:
-            searchPathList.append(importpath[:art_index] + 'shared')
+            search_path_list.append(import_path[:art_index] + 'shared')
 
-        # go through all n_texture search paths
-        for texdir in searchPathList:
+        # go through all texture search paths
+        for texdir in search_path_list:
```

Comment:
This looks so much more complicated than it has to lol

Reply:
Yeah I have an issue which would look to replace this texture path look with a dedicated system, where a user could provide a set of lookup paths.

Reply:
Cant find it be there is a TODO
`# TODO [general][path] Implement full texture path finding.`